src/main/java/com/velox/sloan/cmo/workflows/kapalibrary/SampleToPlateAutoAssigner.java

```diff
@@ -588,26 +609,38 @@ private void addPlateDimensionsToTaskOptions(int plateSize) throws RemoteExcepti
      * @throws com.velox.api.util.ServerException
      */
     private void assignPositionsMapFor96To384WellPlate(List<DataRecord> attachedSamples, String destination384PlateId, List<Map<String, Object>> quadrantValues, String targetPlateIdFieldName, String targetWellIdFieldName, List<DataRecord> newPlateProtocolRecordName) throws NotFound, RemoteException, com.velox.api.util.ServerException {
-        List<Map<String, Object>> newProtocolValues = new ArrayList<>();
         for (Map quadrantMap : quadrantValues) {
             String plateId = quadrantMap.get("PlateId").toString();
+            logInfo("quadrant PlateId: " + plateId);
             String quadrantId = quadrantMap.get("QuadrantNumber").toString();
             for (DataRecord sample : attachedSamples) {
                 String sampleId = sample.getStringVal("SampleId", user);
+                if (sample.getValue("RelatedRecord23", user) == null){
+                    clientCallback.displayError(String.format("Invalid Plate ID '%s' for sample '%s'.\nAll samples must have a Plate ID value for 96 to 384 assignment.",
+                            sample.getValue("RelatedRecord23", user), sampleId));
+                    logError(String.format("Invalid Plate ID '%s' for sample '%s'. All samples must have a Plate ID value for 96 to 384 assignment.",
+                            sample.getValue("RelatedRecord23", user), sampleId));
+                }
                 String samplePlate = sample.getStringVal("RelatedRecord23", user);
+                logInfo("Sample ID: " + sampleId + ", Sample Plates: " + samplePlate);
                 String sampleWell = sample.getStringVal("RowPosition", user).replaceAll("\\s+", "") +
                         sample.getStringVal("ColPosition", user).replaceAll("\\s+", "");
+
                 if (samplePlate.equals(plateId)) {
-                    Map<String, Object> newValues = new HashMap<>();
-                    newValues.put("SampleId", sampleId);
-                    newValues.put(targetPlateIdFieldName, destination384PlateId);
-                    newValues.put(targetWellIdFieldName, to384(Integer.parseInt(quadrantId), sampleWell));
-                    newProtocolValues.add(newValues);
+                    for(DataRecord rec: newPlateProtocolRecordName) {
+                        if (rec.getStringVal("SampleId",user).equals(sampleId)) {
+                            Map<String, Object> newValues = new HashMap<>();
+                            newValues.put("SampleId", sampleId);
+                            newValues.put(targetPlateIdFieldName, destination384PlateId);
+                            newValues.put(targetWellIdFieldName, to384(Integer.parseInt(quadrantId), sampleWell));
+                            logInfo(newValues.toString());
+                            rec.setFields(newValues,user);
```

Comment:
do we need a `break ` is matched sampleId is found ?

Reply:
There will be exactly one sample match, because SampleId's are unique by default, therefore we would not need break statement.